.github/workflows/publish-docs.yaml

```diff
@@ -6,6 +6,9 @@ on:
     branches:
       - main
 
+permissions:
+  contents: write
```

Comment:
IIRC we should add the write permission just for the particular stage of the job that needs it (the deploy docs one) and leave the overall permissions as read

Reply:
The permission can be set for each `Job` but not each `step`.  This workflow has only one job so it may not make difference but I changed the overall permission to `read` and the job permission to `write`.